opentelemetry-sdk/src/opentelemetry/sdk/resources/__init__.py

```diff
@@ -237,7 +266,7 @@ def detect(self) -> "Resource":
         service_name = os.environ.get(OTEL_SERVICE_NAME)
         if service_name:
             env_resource_map[SERVICE_NAME] = service_name
-        return Resource(env_resource_map)
+        return Resource(env_resource_map, "")
```

Comment:
I think you'll need to make changes to the OTELResourceDetector based on the [new additions to the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/sdk.md#detecting-resource-information-from-the-environment):

> Resource detectors that populate resource attributes according to OpenTelemetry semantic conventions MUST ensure that the resource has a Schema URL set to a value that matches the semantic conventions. Empty Schema URL SHOULD be used if the detector does not populate the resource with any known attributes that have a semantic convention or if the detector does not know what attributes it will populate (e.g. the detector that reads the attributes from environment values will not know what Schema URL to use). If multiple detectors are combined and the detectors use different non-empty Schema URL it MUST be an error since it is impossible to merge such resources.

Reply:
I believe that `OTELResourceDetector` is responsible for populating resource attributes from the environment. The example in the spec says that detectors used for this purpose have no way of knowing what Schema URL to use, so leaving it blank seems sensible to me. Please let me know if I'm misunderstanding.

That being said, you reminded me that it's probably the responsibility of the Resource class to validate that the URL is [valid](https://github.com/open-telemetry/oteps/blob/main/text/0152-telemetry-schemas.md#schema-url). I'll go ahead and add that functionality.

Reply:
On second thought, [validating the `schema_url` for every Resource might be excessive and degrade performance.](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/error-handling.md#error-handling-and-performance) I'll put that on the back burner for now.

Reply:
I agree with you that I don't see how OTELResourceDetector would read the schema URL, but the quoted part says _if_ the detector does not know what Schema URL to use, not that it's a given fact. Again, I'm not sure what would need to be changed, since it's a bit ambiguous. Perhaps the schema_url might be specified as an attribute key with value in `OTEL_RESOURCE_ATTRIBUTES` that is supposed to be parsed out? Or there is supposed to be an `OTEL_RESOURCE_SCHEMA_URL`? I think it might be good to double check with the collector community or the otel-python community on CNCF slack?

> Empty Schema URL SHOULD be used if the detector does not populate the resource with any known attributes that have a semantic convention or if the detector does not know what attributes it will populate (e.g. the detector that reads the attributes from environment values will not know what Schema URL to use).

Reply:
The bit in the parentheses is exactly the situation we have here; `OTELResourceDetector` is the detector that reads its attributes from environment values. In that situation, there's no definition for how to communicate a Schema URL to this detector. I'm not sure if that's a feature they want added to the specification.

This does remind me that I need to test to make sure multiple detectors with different Schema URLs generate an error.